lte/gateway/c/session_manager/SessionState.cpp

```diff
@@ -491,11 +491,105 @@ bool SessionState::apply_update_criteria(SessionStateUpdateCriteria& uc) {
   return true;
 }
 
+RuleStats SessionState::get_rule_delta(
+    const std::string& rule_id, uint64_t rule_version, uint64_t used_tx,
+    uint64_t used_rx, uint64_t dropped_tx, uint64_t dropped_rx) {
+  RuleStats ret = {0};
+  auto usage_it = rule_usage_.find(rule_id);
+  // If no previos record found, create one
+  if (usage_it == rule_usage_.end()) {
+    RuleStats s = {0, 0, 0, 0};  //{used_rx, used_rx, dropped_tx, dropped_rx};
```

Comment:
Would be nice to define a function `UsageMap::initialize_rule_stats(const std::string rule_id, const std::string version)` to do all the initialization logic.